src/registry/GithubRegistry.ts

```diff
@@ -72,6 +73,7 @@ export class GithubRegistry extends BaseRegistry implements IRegistry {
     this.repoOwner = pathSegments.at(-2)!;
     this.repoName = pathSegments.at(-1)!;
     this.proxyUrl = options.proxyUrl;
+    this.authToken = options.authToken;
```

Comment:
Should we assert that proxyUrl and authToken can't be set at the same time?

Reply:
actually proxyUrl can still be set, when authentication limit is reached it will fallback to proxyUrl again as the [test here](https://github.com/hyperlane-xyz/hyperlane-registry/pull/506/files#diff-6b6b3cebb717b7cbfec7c6371a8e390788cd6fdf71571e44a1088ea7a783d414R250) testing it

Reply:
Ahh love it